crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -1056,6 +1064,14 @@ where
                     }
                 }
 
+                //dlint
+                if self.settings.rules.enabled(Rule::BadShelveUse) {
+                    dlint::rules::bad_shelve_use(self, stmt);
```

Comment:
Nit: You already now here that you're in a `ImportFrom` so you can write

```
Stmt::ImportFrom(import_from @ ast::StmtImportFrom {
                names,
                module,
                level,
                range: _,
}) => {
	...
	
	if self.settings.rules.enabled(Rule::BadShelveUse) {
         dlint::rules::bad_shelve_use(self, AnyStmtImport::from(stmt));
	}
}
```

Reply:
It does not seem to like this. When I try for example:

```
Stmt::Import(ast::StmtImport { names, range: _ }) => {
...
  //dlint
  if self.settings.rules.enabled(Rule::BadShelveUse) {
         dlint::rules::bad_shelve_use(self, dlint::helpers::AnyStmtImport::from(stmt));
  }
  if self.settings.rules.enabled(Rule::BadMarshalUse) {
         dlint::rules::bad_marshal_use(self, dlint::helpers::AnyStmtImport::from(stmt));
  }
```
Compiler complains:

```
error[E0277]: the trait bound `AnyStmtImport<'_>: std::convert::From<&rustpython_parser::rustpython_ast::Stmt>` is not satisfied
   --> crates/ruff/src/checkers/ast/mod.rs:776:76
    |
776 |                     dlint::rules::bad_shelve_use(self, AnyStmtImport::from(stmt));
    |                                                        ------------------- ^^^^ the trait `std::convert::From<&rustpython_parser::rustpython_ast::Stmt>` is not implemented for `AnyStmtImport<'_>`
    |                                                        |
    |                                                        required by a bound introduced by this call
    |
    = help: the following other types implement trait `std::convert::From<T>`:
              <AnyStmtImport<'a> as std::convert::From<&'a StmtImport>>
              <AnyStmtImport<'a> as std::convert::From<&'a StmtImportFrom>>
```

It seems to want a convert implementation from all of `rustpython_ast::Stmt` even though it's only called in cases of `Stmt` being either `StmtImport` or `StmtImportFrom`. Am I missing anything?

Reply:
Dang, I messed up the code. I'm sorry. 

You have to bind the `StmtImportFrom` in the `match` arm with `import_from @ ast::StmtImportFrom { ... }` and then pass that to the from call:

```rust
Stmt::ImportFrom(import_from @ ast::StmtImportFrom {
                names,
                module,
                level,
                range: _,
}) => {
	...
	
	if self.settings.rules.enabled(Rule::BadShelveUse) {
         dlint::rules::bad_shelve_use(self, AnyStmtImport::from(import_from));
	}
}
```

Reply:
Ah funny I wrote that during trial-and-erroring and my IDE's Rust plugin complained about `use of a moved value` so I didn't build it, but it does seem to work despite those complaints 🤷. Thanks for the feedback and pushed